dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/BaseMetricsRESTProxyServlet.java

```diff
@@ -35,7 +35,7 @@ public abstract class BaseMetricsRESTProxyServlet extends HttpServlet {
 
     /** {@inheritDoc} */
     @Override
-    protected void service(final HttpServletRequest request, final HttpServletResponse response) throws ServletException, IOException {
```

Comment:
I am not sure added `synchronized` will help here as `getAndSetRESTHandlerContainer` and `handleWithDelegate` are already `synchronized`.  Do you know why the session is invalidated? Did the session got invalidated in the code explicitly or due to timeout?